test/api/importcsv/api_test.go

```diff
@@ -0,0 +1,166 @@
+					}
+				}
+			}
+
+			// Check list of Dependencies.
+			gotDeps, _ := Dependency.List()
+			expectedDeps := r.ExpectedDependencies
+			if len(gotDeps) != len(expectedDeps) {
+				t.Errorf("Mismatch in number of imported Dependencies: Expected %d, Actual %d", len(expectedDeps), len(gotDeps))
+			} else {
+				for i, importedDep := range gotDeps {
+					expectedDep := expectedDeps[i].To.Name
+					if importedDep.To.Name != expectedDep {
+						t.Errorf("Mismatch in imported Dependency: Expected %s, Actual %s", expectedDep, importedDep.To.Name)
+					}
+				}
+			}
+
+			// fetch id of CSV file and convert it into required formats
+			var inputID = strconv.FormatUint(uint64(inputData.ID), 10) // to be used for API compatibility
+
+			var outputImportSummaries []api.ImportSummary
+			outputMatchingSummary := api.ImportSummary{}
+			assert.Should(t, Client.Get(api.SummariesRoot, &outputImportSummaries))
+			for _, imp := range outputImportSummaries {
+				if uint(imp.ID) == inputData.ID {
+					outputMatchingSummary = imp
+				}
+			}
+			assert.FlatEqual(len(expectedApps)+len(expectedDeps), outputMatchingSummary.ValidCount)
+
+			// inject import summary id into Summary root
+			pathForImportSummary := binding.Path(api.SummaryRoot).Inject(binding.Params{api.ID: inputID})
+
+			// Get summaries of the Input ID.
+			outputImportSummary := api.ImportSummary{}
+			assert.Should(t, Client.Get(pathForImportSummary, &outputImportSummary))
+
+			// Get all imports.
+			var outputImports []api.Import
+			assert.Should(t, Client.Get(api.ImportsRoot, &outputImports))
+			j, k := 0, 0
+			for _, imp := range outputImports {
+				if imp["recordType1"] == 1 && j < len(expectedApps) {
+					// An Application with no dependencies.
+					if expectedApps[j].Name != imp["applicationName"] {
+						t.Errorf("Mismatch in name of import: Expected %s, Actual %s", expectedApps[j].Name, imp["applicationName"])
+					}
+					if expectedApps[j].Description != imp["description"] {
+						t.Errorf("Mismatch in name of import: Expected %s, Actual %s", expectedApps[j].Description, imp["description"])
+					}
+					if expectedApps[j].BusinessService.Name != imp["businessService"] {
+						t.Errorf("Mismatch in name of import: Expected %s, Actual %s", expectedApps[j].BusinessService.Name, imp["businessService"])
+					}
+					j++
+				}
+				if imp["recordType1"] == 2 && k < len(expectedDeps) {
+					// An Application with Dependencies.
+					if expectedDeps[k].From.Name != imp["applicationName"] {
+						t.Errorf("Mismatch in name of import: Expected %s, Actual %s", expectedDeps[k].From.Name, imp["applicationName"])
+					}
+					if expectedDeps[k].To.Name != imp["dependency"] {
+						t.Errorf("Mismatch in name of import: Expected %s, Actual %s", expectedDeps[k].To.Name, imp["dependency"])
+					}
+					k++
+				}
+			}
+
+			// Download the csv.
```

Comment:
Not a blocker, the Download CSV test part looks OK, but I don't think it should be part of the import test since it actually doesn't import anything and is kind of independent on most of code in this test method. Would you mind move it to separate test method like TestDownloadCSV?

Reply:
Not a problem, keep it as it is.